File: 1768-merge-strings-alternately/1768-merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
    result := ""
	word1Pointer := 0
	word2Pointer := 0

	for word1Pointer < len(word1) && word2Pointer < len(word2) {
		result += string(word1[word1Pointer])
		result += string(word2[word2Pointer])

		word1Pointer++
		word2Pointer++
	}

	for word1Pointer < len(word1) {
		result += string(word1[word1Pointer])
		word1Pointer++
	}

	for word2Pointer < len(word2) {
		result += string(word2[word2Pointer])
		word2Pointer++
	}

	return result
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int ptr1 = 0;
        int ptr2 = 0;

        String mergedResult = "";
        boolean isAddWord1 = true;
        while (ptr1 < word1.length() && ptr2 < word2.length()){
            if (isAddWord1){
                mergedResult += word1.charAt(ptr1);
                ptr1++;
            }
            else{
                mergedResult += word2.charAt(ptr2);
                ptr2++;
            }
            isAddWord1 = !isAddWord1;
        }

        while (ptr1 < word1.length()){
            mergedResult += word1.charAt(ptr1);
            ptr1++;
        }

        while (ptr2 < word2.length()){
            mergedResult += word2.charAt(ptr2);
            ptr2++;
        }

        return mergedResult;
    }
}